Visitor knows when someone hear him

gturri opened the pull request:
Some code has been copy / pasted. So even if you're interested in this feature you may want to wait for some cleaning before pulling it.
However I'm creating this pull request now in case you already have some feedback.

I believe I could add a few commits to this pull request to:
- Add timestamps to message (because it's nice to see that someone is coming (or ringing), but it's nice to see if it's just now or if it happened an hour ago)
- Send feedback (like "the bell is ringing" to the client, when he clicked, to let him know if no one heard because of his 3G)
- Remove code duplication

AlexFaraino commented on ring.html:
```diff
@@ -20,11 +21,29 @@
     };
+
+    // I'm a client
+    socket.emit('type', { type: 'client'});
```
I'd prefer visitor (or guest) to client. "Host" is not here for server, it's the person who own the place, and welcomes visitors or guests :)

AlexFaraino commented:
O.K. for the timestamp. That's a good idea.

About the ACK: I don't think the visitor interface should display an history of messages. Instead, can we use a simple input (like the one for monitoring connection on the /monitor page)? When someone rings, it displays "Ringing..." and when an host ACK, it displays "Someone is coming".

What do you think ?

gturri commented:
"What do you think ?"
Not much: I'm not really a IHM guy.

However, you're right: a visitor isn't interested in the history. Anyway he is supposed to open the app when he arrives, and close it quite soon after.

I'll try to adapt this PR.

gturri committed:
Send a feedback to the visitor when it's actuall ringing

It's currently to every visitor currently connected because it was the most straightforward
to implement, and it shouldn't hurt

gturri committed:
The host can say he's coming

gturri committed:
Timestamp messages

gturri committed:
Add some debug logs

gturri commented:
Here is another proposition.

I believe next step would be removing the "ringing" message after 10s (so that the visitor doesn't believe it's ringing non-stop)

AlexFaraino commented:
I agree !

gturri committed:
Don't print "it's ringing" forever

I'm not applying this to the "I'm coming message" because it's normal that
this action last for some long minutes.

gturri commented:
Here it is, here is what I add in mind regarding this feature.

Please let me know if you have any feedback, or if you'd rather have it done differently.
